### PR TITLE
Implement favorites without adding tests

### LINKED ISSUE #69: Implement favorites feature
As a user, I can "favourite" a frequent contact so that I don't have to keep searching for them whenever I need their information.

### PR DESCRIPTION
**Pull request by jugheadjones10**
Closes #69

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Person.java [hunk 1]
```diff
@@ -14,19 +14,18 @@
...
     private final Email email;
-
     // Data fields
     private final Address address;
     private final Rank rank;
     private final Unit unit;
     private final Company company;
     private final Platoon platoon;
     private final Set<Tag> tags = new HashSet<>();
+    private Boolean isFavorite = false;
```
Comment (rexcyrio) [inline src/main/java/seedu/address/model/person/Person.java:28]: See comment on wrapper class `Boolean`

File: src/main/java/seedu/address/model/person/Person.java [hunk 2]
```diff
@@ -85,6 +84,14 @@ public Set<Tag> getTags() {
         return Collections.unmodifiableSet(tags);
     }
 
+    public Boolean getIsFavorite() {
```
Comment (rexcyrio) [inline src/main/java/seedu/address/model/person/Person.java:87]: Likewise, see comment on wrapper class `Boolean`

**Review by rexcyrio — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/logic/commands/FavoriteCommand.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        if (!personToFavorite.isSamePerson(editedPerson) && model.hasPerson(editedPerson)) {
+            throw new CommandException(MESSAGE_DUPLICATE_PERSON);
+        }
+
+        model.setPerson(personToFavorite, editedPerson);
+        model.updateFavoritedPersonList();
+        model.commit(model.getAddressBook());
+
+
+        model.updateFavoritedPersonList();
```
[the remark this answers is not in the document]
Comment (rexcyrio) [reply src/main/java/seedu/address/logic/commands/FavoriteCommand.java]: This has been resolved in the commit "Add tests for favorite command".

File: src/main/java/seedu/address/model/Model.java [hunk 4]
```diff
@@ -104,10 +105,22 @@ public interface Model {
...
      * @throws NullPointerException if {@code predicate} is null.
      */
     void updateFilteredPersonList(Predicate<Person> predicate);
+
+    /**
+     * Updates the filter of the favorited person list to filter by the latest favorites.
+     *
+     * @throws NullPointerException if {@code predicate} is null.
+     */
+    void updateFavoritedPersonList();
```
[the remark this answers is not in the document]
Comment (rexcyrio) [reply src/main/java/seedu/address/model/Model.java]: This has been resolved in the commit "Add tests for favorite command".

File: src/main/java/seedu/address/ui/FavoritesListPanel.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        protected void updateItem(Person person, boolean empty) {
+            super.updateItem(person, empty);
+
+            if (empty || person == null) {
+                setGraphic(null);
+                setText(null);
+            } else {
+                int filteredListPersonIndex = filteredPersonList.indexOf(person);
+                System.out.println("Filtered person index: " + filteredListPersonIndex);
+                setGraphic(new PersonCard(person, filteredListPersonIndex + 1).getRoot());
```
[the remark this answers is not in the document]
Comment (jugheadjones10) [reply src/main/java/seedu/address/ui/FavoritesListPanel.java:47]: Fixed this as part of fixing a similar bug discovered in PED

**Comment by rexcyrio** [comment]
Comments that are currently not yet resolved:
- https://github.com/AY2223S2-CS2103T-W10-3/tp/pull/78#discussion_r1148523979
- https://github.com/AY2223S2-CS2103T-W10-3/tp/pull/78#discussion_r1148527472

**Pull request merged**